.grit/patterns/moment-to-date-fns.grit

```diff
@@ -0,0 +1,358 @@
+language js
+
+// A call to `add`/`sub` method of a `momentjs` date object.
+pattern moment_arith($date, $method, $count, $pat) {
+  `$date.$method($count, $pat)` as $arith_exp where {
+    $method <: or { `add`, `sub` },
+    if ($method <: `add`) {
+      $method <: ensure_import_from(source=`"date-fns/add"`),
+    } else {
+      $method <: ensure_import_from(source=`"date-fns/sub"`),
+    },
+    $pat <: string()
+  }
+}
+
+// A call to `add/sub` where the caller (`this`) is an identifier. eg: foo.add(bar);
+pattern moment_arith_lhs_identifier($date, $method, $count, $pat) {
+  moment_arith($date, $method, $count, $pat) where {
+    $date <: identifier()
+  }
+}
+
+// A `const` variable declaration that initializes a moment date object
+pattern moment_const_decl($declaration, $declarators) {
+  `$varName = $moment()` where {
+    $moment <: or { `moment`, `$_.duration` },
+    $varName <: within `$declaration` where { $declaration <: `const $declarators` },
+  }
+}
+
+// Matches calls like `moment.startOf('day')`, `moment.startOf('week')`
+pattern moment_set_to_start_or_end_literal($date, $func_type, $unit) {
+  `$date.$func_type($literal)` where {
+    or { $func_type <: `startOf`, $func_type <: `endOf` },
+    $literal <: string(fragment=$unit)
+  }
+}
+
+// moment.max(...) or moment.min(...)
+pattern moment_limit_to_max_or_min($date, $arg, $methodName) {
+  `$date.$methodName($arg)` where {
+    or { $methodName <: "max", $methodName <: "min" }
+  }
+}
+
+pattern moment_set_or_get($duration_or_date, $accessor, $accessor_fn, $arg) {
+   `$duration_or_date.$accessor($arg)` where {
+    $accessor <: r"(year|month|week|day|hour|minute|second|millisecond)s?"($unit),
+    $capitalized_unit = capitalize(string=$unit),
+    $accessor_fn = join(list=["get", $capitalized_unit, "s"], separator=""),
```

Comment:
@srijan-paul We pushed an update so you can use braces: https://docs.grit.io/language/patterns#metavariables

```suggestion
    $accessor_fn = `get$[capitalized_unit]s`
```